### DISCUSSION TITLE
PCPCombine: Period Max/Min of -add/-substract fields

### DISCUSSION THREAD

**Discussion by jtv108-wx** [dtcenter/METplus#2170] [Configuration] [answered]
Hi,

Is it possible (with one PCPcombine run) to use PCPCombine to get a 24-hr maximum field of a pair of variables that have been substracted over a 1-hr interval in the period? Also, could this 24-hr max field be multiplied by some constant value?

As an example, I'd like to use PCPCombine and Ensemble_STAT to calculate a 24-hr NMEP of the LCL heights from ensemble data (using 2m temperature and 2m dew points). I did see that using "WIND" as a FCST_PCP_COMBINE_INPUT_NAME takes UGRD and VGRD (at a set input level) to calculate speed. Are there any other FCST_PCP_COMBINE_INPUT_NAME inputs that read multiple variables at once (LCL, STP, etc...)?

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. If I'm understanding correctly, you want to take two separate fields, subtract them in 1 hour intervals, and then find the maximum of those two fields over a 24 hour period? If so, PCPCombine does not have the current capability to solve that issue in one step.

You can accomplish this over multiple calls to PCPCombine: for example, you can request the maximum of two fields, then pass the results back to a new call of PCPCombine to subtract the two maximum fields. You could also do the opposite direction (subtract the two fields every hour, then pass the resulting field back to PCPCombine for a 24 hour maximum). 

You can also utilize a Python script that is called by PCPCombine which can take over one of these calls and complete the task in 1 step: for example, your Python embedding script can compute the difference of the two fields, which is then passed to PCPCombine to compute the 24 hour maximum of the field. That would require some additional coding on your end but would complete the task in 1 call to PCPCombine.

Finally, to your question 

> ...could this 24-hr max field be multiplied by some constant value

 I'm happy to say that's an easier task. PCPCombine allows the use of the "convert" function that can utilize set mathematical operations to a single field. the "convert" option is available in the field string. An example of this would be

```
-field ‘name=”ACPCP”; level=”A6”; convert(x)=x/25.4;’
```

Where the 6-hour ACPCP fields are converted from millimeters to inches. [Use this link](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#fcst) (you'll need to scroll down a bit) to find more information on what the "convert" function is capable of.

**Reply by jtv108-wx** [reply]
Hi,

That is exactly what I was attempting to do using PCPCombine. I'll give your suggestion a try using multiple calls to PCPCombine. 

Great, I'll take a look at the link for using the convert(x) function.

Thanks for the help.

**Reply by j-opatz** [reply]
Hi there,

I just wanted to check in and see if you were able to make the PCPCombine method work for this problem.
